lib/Service/GitlabAPIService.php

```diff
@@ -448,7 +448,12 @@ public function request(?string $userId, string $endPoint, array $params = [], s
 		if ($userId !== null) {
 			$this->checkTokenExpiration($userId);
 		}
-		$baseUrl = $this->getConnectedGitlabUrl($userId);
+		$adminOauthUrl = $this->config->getAppValue(Application::APP_ID, 'oauth_instance_url', Application::DEFAULT_GITLAB_URL) ?: Application::DEFAULT_GITLAB_URL;
+		if ($userId === null) {
+			$baseUrl = $adminOauthUrl;
+		} else {
+			$baseUrl = $this->config->getUserValue($userId, Application::APP_ID, 'url', $adminOauthUrl) ?: $adminOauthUrl;
+		}
```

Comment:
this makes use of the user's `url` config value which works if the user did not change it. If changed, all the requests are invalid.

The point I'm making is the user's `url` config does not serve any purpose after the `oauthRedirect` function overrides it and will lead to confusion.